dask/dataframe/core.py

```diff
@@ -1452,12 +1452,26 @@ def var(self, axis=None, skipna=True, ddof=1, split_every=False, dtype=None, out
             return handle_out(out, result)
         else:
             num = self._get_numeric_data()
-            x = 1.0 * num.sum(skipna=skipna, split_every=split_every)
-            x2 = 1.0 * (num ** 2).sum(skipna=skipna, split_every=split_every)
-            n = num.count(split_every=split_every)
-            name = self._token_prefix + 'var'
-            result = map_partitions(methods.var_aggregate, x2, x, n,
-                                    token=name, meta=meta, ddof=ddof)
+            values_dtype = num.values.dtype
+
+            if not np.issubdtype(values_dtype, np.number):
+                num.values.dtype = 'f8'
```

Comment:
What does this do? Do we want `.astype` instead?

Alos, this may mutate the data inplace? I think that `num.values` may be a view on the data backing the DataFrame? Not sure.

Reply:
@TomAugspurger It was added to make sure dtype is not 'object' (if array contains booleans and timedeltas). I think I could avoid this, I'll look into it, thank you for the comment.